benchmark/utils/DriverUtils.swift

```diff
@@ -167,28 +161,23 @@ struct TestConfig {
   }
 
   mutating func findTestsToRun() {
-    var i = 1
-    for benchName in precommitTests.keys.sorted() {
-      tests.append(Test(name: benchName, n: i, f: precommitTests[benchName]!))
-      i += 1
-    }
-    for benchName in otherTests.keys.sorted() {
-      tests.append(Test(name: benchName, n: i, f: otherTests[benchName]!))
-      i += 1
-    }
-    for benchName in stringTests.keys.sorted() {
-      tests.append(Test(name: benchName, n: i, f: stringTests[benchName]!))
-      i += 1
-    }
-    for i in 0..<tests.count {
-      if onlyPrecommit && precommitTests[tests[i].name] == nil {
-        tests[i].run = false
-      }
-      if !filters.isEmpty &&
-         !filters.contains(String(tests[i].index)) &&
-         !filters.contains(tests[i].name) {
-        tests[i].run = false
+    let allTests = [precommitTests, otherTests, stringTests]
+      .map { dictionary -> [(key: String, value: (Int)-> ())] in
+        Array(dictionary).sorted { $0.key < $1.key } } // by name
+      .joined()
+
+    let included: Set<String>
+      if !filters.isEmpty  { included = Set(filters) } else {
+        if onlyPrecommit { included = Set(precommitTests.keys) }
+        else { included = Set(allTests.map { $0.key }) }
       }
+
```

Comment:
No I don't want `enumerated`, because it is off by one. 

Again, this is something that lead me to file [SR-4746 Let enumerated() count from specified number](https://bugs.swift.org/browse/SR-4746), pitch [swift-evolution proposal](https://github.com/palimondo/swift-evolution/blob/sequence-cleanup/proposals/NNNN-enumerate-from.md)  but @airspeedswift says [`zip` is the way to go](https://lists.swift.org/pipermail/swift-evolution/Week-of-Mon-20170501/036361.html). Rob Napier also independently [refactored this code to use of `zip`](https://gist.github.com/rnapier/e5a9c883b37fb65cb412faf5da7f9e9c) when I asked him to improve on my style. 

I don't normally summon the [Council of Elrond](http://lotr.wikia.com/wiki/Council_of_Elrond) about single bug fix, but this one irritated me so much I got my [first article out of it](https://medium.com/@palimondo/kick-your-manual-iteration-habit-a75a6b3cfa8e)...